testbook/testbook.py

```diff
@@ -1,15 +1,30 @@
-import nbformat
+import functools
+from unittest.mock import DEFAULT
 
-try:
-    from pytest import fixture
-except ImportError:  # pragma: no cover
-    def fixture(func, *args, **kwargs):
-        return func
+import nbformat
 
 from .client import TestbookNotebookClient
 
 
 class testbook:
+    """`testbook` acts as function decorator or a context manager.
+
+    When the function/with statement exits the kernels started when
+    entering the function/with statement will be terminated.
+
+    If `testbook` is used as a decorator, the `TestbookNotebookClient`
+    will be passed as first argument to the decorated function.
+    """
+    # Developer notes:
+    #
+    # To trick pytest, we mimic the API of unittest.mock.patch in testbook.
+    # Notably, the following elements are added:
+    # * attribute_name, Class attribute (see below)
+    # * new, Instance attribute (see __init__)
+    # * patchings, wrapper attributes (see __call__)
```

Comment:
I'm not sure I fully understand how this is working. Can you please elaborate a little further? I just want to wrap my head around this before I approve it. 

cc @MSeal

Reply:
The only case I know of a decorator that is playing well with pytest is `unittest.mock.patch`. Looking at the code of it and the code of pytest, I figure out that pytest has specific handler for that decorator on test functions.
So to play nicely with pytest, this PR trick pytest so it thinks `testbook` is `unittest.mock.patch` and it serves correctly the client in the test function.

The other solution would be to change the signature of the test function decorated by testbook in order to remove the client from the input arguments because pytest uses signature introspection to figure out which fixture are needed by a specific test. This will be quite complicated.

Another solution is to not support the decorator for pytest tests.

Reply:
Thanks for the elaborate explanation!